### PR TITLE
Add copy support for cube/axis/column names.

### PR DESCRIPTION
**Pull request by tympollack**

### CODE REVIEW

File: src/main/resources/static/js/ntwobe.js [hunk 1]
```diff
@@ -3095,47 +3095,48 @@ var NCubeEditor2 = (function ($) {
...
-                    clipData += '"' + content.replace(/"/g, '""') + '"';
-                } else {
-                    clipData += content;
-                }
+                    if (content.indexOf('\n') > -1) {
+                        // Must quote if newline (and double any quotes inside)
+                        clipData += '"' + content.replace(/"/g, '""') + '"';
+                    } else {
+                        clipData += content;
+                    }
```
Comment (gaigstroliac) [inline src/main/resources/static/js/ntwobe.js]: ternary?

```javascript
clipData += (content.indexOf('\n') > -1) ? ('"' + content.replace(/"/g, '""') + '"') : content;
```

File: src/main/resources/static/js/ntwobe.js [hunk 2]
```diff
@@ -3144,24 +3145,75 @@ var NCubeEditor2 = (function ($) {
...
+
+        function getAxisNames(axisNumStart, axisNumEnd) {
+            var i, len;
+            var vals = '';
+            for (i = axisNumStart, len = axisNumEnd; i <= len; i++) {
+                if (vals) {
+                    vals += '\n';
+                }
+                vals += axes[i].name;
+            }
```
Comment (gaigstroliac) [inline src/main/resources/static/js/ntwobe.js]: I can't recall if I've ever checked to efficiency of it, but for stuff like this, I often find it easier to add items to an array and then return with a `join`. That way you don't have to constantly do a check for the last item.

```javascript
var vals = [];
for (i = axisNumStart, len = axisNumEnd; i <= len; i++) {
  vals.push(axes[i].name);
}
return vals.join('\n');
```

for even more terse and readable, but at the likely expense of efficiency
```javascript
  return axes
    .slice(axisNumStart, asixNumEnd+1)
    // I would so like to write the next line as .map(axis => axis.name)
    .map(function(axis){ 
      return axis.name; 
    })
    .join('\n');
```

**Comment by gaigstroliac** [comment]
Looks like some of my comments were only about things you indented, so take them for what they're worth.

File: src/main/resources/static/js/ntwobe.js (hunk 2 above)
Comment (tympollack) [inline src/main/resources/static/js/ntwobe.js]: That's true about array join, I do this a lot and never thought about using that...

As for slice, map, join chain, I'll have to check efficiency, but does seem more readable.

File: src/main/resources/static/js/ntwobe.js (hunk 1 above)
Comment (tympollack) [inline src/main/resources/static/js/ntwobe.js]: Maybe, looks kinda messy tho.

**Comment by tympollack** [comment]
@gaigstroliac yeah the bulk of it was wrapping existing code into an if, so it's not all new.

**Pull request merged**